### ISSUE TITLE
Proposition: Remplace "scalar" keyword by the scalar types on PhpDocs

### ISSUE THREAD

**Issue by JesusValera** [jenshaase/phel-lang#119]
### TL;DR: I proposed to replace all `scalar` keywords to `string|float|int|bool`. ⚠️ 

# 🔗 Description

As the PHP [documentation](https://www.php.net/manual/en/language.types.intro.php) says, there are four _scalar_ types in PHP:
* boolean
* integer
* float
* string

Also, there is a [`is_scalar()`](https://www.php.net/manual/en/function.is-scalar.php) method, but the `scalar` type doesn't exist per see. 🎯 

---
### Example 1 🌜 

I attach one picture of PHPStorm from this project which the `scalar` keyword appears highlight, which means it doesn't exist, and bottom the same method with the "correct" implementation.

![Selection_157](https://user-images.githubusercontent.com/6381924/95131280-11b00280-075e-11eb-80aa-4ca936795f2d.jpg)
---

### Example 2 🌛 

You can see in this second picture the PhpDoc returns an `AbstractType, null and scalar`, the method itself (L115) returns a `string` (but string is not a "scalar"), the program works as expected, but sometimes you see this kind of IDE "warnings/error/whatever".

![Selection_158](https://user-images.githubusercontent.com/6381924/95131295-170d4d00-075e-11eb-9340-7adf0fea08e2.jpg)

What do you think? 💯 

----

>**Pros:**
>- IDE help
>- No annoying messages
>
>**Cons:**
>- PhpDoc too long (which is not a real problem in my opinion)

**Comment by jenshaase** [comment]
Yes we can do this. Just create a pull request and I can merge it!

**Comment by jenshaase** [comment]
Fixed with PR #120
